In [124]:
ls

 Volume in drive C has no label.
 Volume Serial Number is 372F-3690

 Directory of C:\Users\USER\Documents\Py_Project\RTA

03/17/2018  09:51 PM    <DIR>          .
03/17/2018  09:51 PM    <DIR>          ..
03/17/2018  08:19 PM    <DIR>          .ipynb_checkpoints
03/17/2018  08:14 PM           129,725 CSDL_KhachHang.xlsx
03/17/2018  09:51 PM           163,614 rtAnalyst.ipynb
               2 File(s)        293,339 bytes
               3 Dir(s)  57,840,959,488 bytes free


In [125]:
import numpy as np
import pandas  as pd
%matplotlib inline

## #1 OUTPUT CAN BE EXCEL OR CSV FILE
### =>>> Alright! I'll still use .xlsx format so i dont need to change to .csv anymore

In [126]:
data=pd.read_excel('CSDL_KhachHang.xlsx')

In [127]:
data.head(50)

STT                     Tên KH      Mã KH       Ngày nhập CSDL  \
0     1             Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
1     2                    Bà Tiến     CP.022  2017-01-23 12:38:01   
2     3                    Bà Tĩnh     CP.018  2017-01-23 12:38:01   
3     4                     Bà Trừ     CP.056  2017-01-23 12:38:01   
4     5                Bùi Đức Hải     TC.127  2017-01-23 12:38:01   
5     6               Bùi Đức Lĩnh     TC.130  2017-01-23 12:38:01   
6     7            Bùi Quang Thành     TC.155  2017-01-23 12:38:01   
7     8         Bùi Thị Liên(Dũng)     AD.004  2017-01-23 12:38:01   
8     9               Bùi Thị Thao     TC.014  2017-01-23 12:38:01   
9    10               Bùi Văn Đăng     TC.086  2017-01-23 12:38:01   
10   11                Bùi Văn Đạo     TC.045  2017-01-23 12:38:01   
11   12              Bùi Văn Nghĩa     TC.032  2017-01-23 12:38:01   
12   13              Bùi Văn Thang     TC.019  2017-01-23 12:38:01   
13   14              Bùi Văn Thông     TC.018  2017-01-23 12:38:01   
14   15                Bùi Văn Tửu     TC.021  2017-01-23 12:38:01   
15   16        Bùi Văn Tuyên(hinh)     TC.031  2017-01-23 12:38:01   
16   17                Cao Thị Lơi     TB.163  2017-01-23 12:38:01   
17   18         Công Ty Đại Việt 1     LM.052  2017-01-23 12:38:01   
18   19          Công Ty Pháp Việt     LM.053  2017-01-23 12:38:01   
19   20              Đàm Hữu Thành     LM.046  2017-01-23 12:38:01   
20   21                 Đặng Bá Bạ     CP.008  2017-01-23 12:38:01   
21   22            Đặng Quang Nghị     XA.056  2017-01-23 12:38:01   
22   23           Đặng Quang Tuyên     ĐX.014  2017-01-23 12:38:01   
23   24  Đặng Quang Tuyến ( Trang)     ĐX.001  2017-01-23 12:38:01   
24   25               Đặng Thị Bắc     XA.069  2017-01-23 12:38:01   
25   26               Đặng Thị Keo     XA.014  2017-01-23 12:38:01   
26   27             Đặng Thị Tuyên     XA.079  2017-01-23 12:38:01   
27   28           Đặng Trung Tuyến     XA.110  2017-01-23 12:38:01   
28   29                Đặng Văn Bộ  ĐX.025.01  2017-01-23 12:38:01   
29   30              Đặng Văn Động     CP.003  2017-01-23 12:38:01   
30   31              Đặng Văn Dũng     XA.030  2017-01-23 12:38:01   
31   32               Đặng Văn Duy     XA.125  2017-01-23 12:38:01   
32   33              Đặng Văn Huân     XA.124  2017-01-23 12:38:01   
33   34               Đặng Văn Ích     XA.038  2017-01-23 12:38:01   
34   35      Đặng Văn Lịch ( Thắm)     XA.130  2017-01-23 12:38:01   
35   36              Đặng Văn Sáng     XA.059  2017-01-23 12:38:01   
36   37               Đặng Văn Sao     XA.020  2017-01-23 12:38:01   
37   38       Đặng Văn Sơn ( Tuyền     XA.009  2017-01-23 12:38:01   
38   39               Đặng Văn Tại     ĐX.040  2017-01-23 12:38:01   
39   40              Đặng Văn Tiếp     XA.123  2017-01-23 12:38:01   
40   41             Đặng Văn Trọng     XA.012  2017-01-23 12:38:01   
41   42             Đặng Văn Tưởng     XA.100  2017-01-23 12:38:01   
42   43          Đào Huy Thế(hạnh)     ĐX.056  2017-01-23 12:38:01   
43   44              Đào Quốc Công     XA.066  2017-01-23 12:38:01   
44   45         Đào Văn Phan ( Tỳ)     AD.050  2017-01-23 12:38:01   
45   46              Đào Văn Thanh     TG.039  2017-01-23 12:38:01   
46   47              Đào Văn Thành     AD.008  2017-01-23 12:38:01   
47   48              Đào Văn Thành     XA.073  2017-01-23 12:38:01   
48   49               Đào Xuân Tựu     TC.038  2017-01-23 12:38:01   
49   50       Đinh Huy Cường ( Vải  TG.105.03  2017-01-23 12:38:01   

           Thôn           Xã      Huyện      Tỉnh Mã thôn  Mã xã  Mã huyện  \
0     Thanh Gia    Quảng Phú  Lương Tài  Bắc Ninh      TG   9514       264   
1     Chinh Phú  Trung Chính  Lương Tài  Bắc Ninh      CP   9523       264   
2     Chinh Phú  Trung Chính  Lương Tài  Bắc Ninh      CP   9523       264   
3     Chinh Phú  Trung Chính  Lương Tài  Bắc Ninh      CP   9523       264   
4   Trung Chinh  Trung Chính  Lương Tài  Bắ

## #2 HOUSEHOLDS MUST HAVE A VALID PHONE NUMBERS

In [128]:
PhoneNumber = data['SĐT']

In [129]:
data=data[PhoneNumber>0]
data

STT                    Tên KH      Mã KH       Ngày nhập CSDL  \
0        1            Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
50      51             Đinh Huy Dũng  TG.105.02  2017-03-11 08:50:23   
52      53            Đinh Thị Hường     TG.141  2017-03-11 09:00:04   
54      55              Đinh Văn Anh  TG.105.01  2017-03-10 16:48:46   
57      58       Đinh Văn Ánh ( Đan)  TG.071.06  2017-03-10 11:02:33   
58      59             Đinh Văn Hanh  TG.076.01  2017-03-10 11:04:28   
70      71             Đinh Văn Phẩm     TG.077  2017-03-10 11:07:01   
74      75            Đinh Văn Quyền     TG.059  2017-03-10 10:39:49   
79      80   Đinh Văn Trường (luyện)     TG.037  2017-03-10 10:29:22   
81      82               Đinh Văn Vu     TG.103  2017-03-10 16:46:55   
91      92         Đỗ Đình Đề ( Minh     TB.251  2017-03-02 09:17:11   
94      95       Đỗ Đình Đông ( Thủy     TB.227  2017-02-17 09:59:58   
98      99                Đỗ Đình Lệ     TB.049  2017-02-28 10:21:05   
100    101       Đỗ Đình Luận ( Thúy     TB.226  2017-02-17 09:57:11   
106    107             Đỗ Đình Ngoạn     LM.002  2017-02-27 15:01:09   
107    108         Đỗ Đình Phú ( Năm     TB.248  2017-03-02 09:08:45   
109    110            Đỗ Đình Phương  TB.010.04  2017-02-28 10:13:02   
111    112             Đỗ Đình Quang     TB.241  2017-02-24 10:13:35   
114    115                Đỗ Đình Sở  TB.017.02  2017-02-28 10:14:51   
122    123         Đỗ Đình Tộ ( Loan     TB.252  2017-03-02 09:18:57   
137    138           Đỗ Thị Sợi(Hữu)     AP.117  2016-12-13 09:57:14   
146    147              Đỗ Văn Thắng  LM.012.01  2017-02-27 15:06:41   
173    174           Đoàn Thị Phượng     AP.039  2016-12-13 09:51:33   
207    208  Hoàng Văn Chiến ( Thanh)     TG.146  2017-03-11 09:01:16   
208    209             Hoàng Văn Chữ     TG.145  2017-03-11 09:00:45   
215    216       Hoàng Văn Nam (ước)     TG.101  2017-03-10 16:46:20   
216    217  Hoàng Văn Nghĩa ( Tuyết)     TG.134  2017-03-11 08:58:49   
217    218       Hoàng Văn Nhã ( Hào     TG.132  2017-03-11 08:58:14   
376    377         Ngô Thị Dung ( Vỹ     TB.187  2017-02-28 10:25:27   
378    379       Ngô Thị Thảo (ngọc)     TB.051  2017-02-28 10:21:41   
...    ...                       ...        ...                  ...   
1310  1311             Trịnh Văn Đắp     TB.006  2017-02-28 10:10:14   
1311  1312             Trịnh Văn Đậu  TB.042.01  2017-02-28 10:19:52   
1319  1320      Trịnh Văn Dư (luyến)     TG.080  2017-03-10 11:08:12   
1321  1322    Trịnh Văn Dụng ( Tuyến     TB.243  2017-02-24 10:18:47   
1328  1329            Trịnh Văn Hiệu     TB.018  2017-02-28 10:15:40   
1332  1333   Trịnh Văn Hồi ( Lan ) 2     TB.247  2017-02-25 11:09:22   
1337  1338     Trịnh Văn Huân ( Sang  TG.085.01  2017-03-10 16:43:03   
1338  1339             Trịnh Văn Huê  TB.001.05  2017-02-28 10:08:35   
1340  1341       Trinh Văn Huệ ( Tâm     TB.236  2017-02-24 10:00:27   
1347  1348      Trịnh Văn Khởi ( Quý     TB.215  2017-02-28 10:26:37   
1354  1355              Trịnh Văn Lự  TG.093.01  2017-03-10 16:43:43   
1355  1356      Trịnh Văn Luân ( Huệ     TG.127  2017-03-11 08:57:01   
1363  1364    Trịnh Văn Mạnh ( Phong  TB.028.07  2017-02-28 10:17:30   
1367  1368       Trịnh Văn Năm( Phúc     LM.083  2017-02-28 09:51:51   
1376  1377    Trịnh Văn Phương ( ngũ     TB.229  2017-02-17 10:27:18   
1380  1381      Trịnh Văn Quý ( Sạnh     TB.245  2017-02-25 10:58:18   
1383  1384      Trịnh Văn Soi ( Sóng  TB.023.02  2017-02-28 10:16:13   
1385  1386             Trịnh Văn Tâm     TB.003  2017-02-28 10:09:07   
1386  1387     Trịnh Văn Tấn ( bố Tạ     TB.249  2017-03-02 09:10:57   
1393  1394   Trịnh Văn The ( Dung) 2     TB.246  2017-02-25 11:05:11   
1396  1397     Trịnh Văn Thi (chiêm)     TB.017  2017-02-28 10:13:53   
1403  1404            Trịnh Văn Tiền     TB.047  2017-02-28 10:20:30   
1410  1411          Trịnh Văn Trường  TG.122.01  2017-03-11 08:56:25   
1413  1414    Trịnh Văn Tuấn ( Chinh     TB.225

##### In your dataset, All Phone numbers are mobile phone which i just need to include '0' at the beginning of these numbers. There's no land-line so i dont need to include zone code (like 024 or 028) here

In [130]:
AfterConvert=[]
for i in data['SĐT']:
    i=i.astype(str)
    k=i.replace(i,('0'+i))
    #j=pd.to_numeric(k, errors='coerce')
    AfterConvert.append(k)

In [131]:
data.insert(5,'SĐT Mới',value=AfterConvert)

In [132]:
data

STT                    Tên KH      Mã KH       Ngày nhập CSDL  \
0        1            Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
50      51             Đinh Huy Dũng  TG.105.02  2017-03-11 08:50:23   
52      53            Đinh Thị Hường     TG.141  2017-03-11 09:00:04   
54      55              Đinh Văn Anh  TG.105.01  2017-03-10 16:48:46   
57      58       Đinh Văn Ánh ( Đan)  TG.071.06  2017-03-10 11:02:33   
58      59             Đinh Văn Hanh  TG.076.01  2017-03-10 11:04:28   
70      71             Đinh Văn Phẩm     TG.077  2017-03-10 11:07:01   
74      75            Đinh Văn Quyền     TG.059  2017-03-10 10:39:49   
79      80   Đinh Văn Trường (luyện)     TG.037  2017-03-10 10:29:22   
81      82               Đinh Văn Vu     TG.103  2017-03-10 16:46:55   
91      92         Đỗ Đình Đề ( Minh     TB.251  2017-03-02 09:17:11   
94      95       Đỗ Đình Đông ( Thủy     TB.227  2017-02-17 09:59:58   
98      99                Đỗ Đình Lệ     TB.049  2017-02-28 10:21:05   
100    101       Đỗ Đình Luận ( Thúy     TB.226  2017-02-17 09:57:11   
106    107             Đỗ Đình Ngoạn     LM.002  2017-02-27 15:01:09   
107    108         Đỗ Đình Phú ( Năm     TB.248  2017-03-02 09:08:45   
109    110            Đỗ Đình Phương  TB.010.04  2017-02-28 10:13:02   
111    112             Đỗ Đình Quang     TB.241  2017-02-24 10:13:35   
114    115                Đỗ Đình Sở  TB.017.02  2017-02-28 10:14:51   
122    123         Đỗ Đình Tộ ( Loan     TB.252  2017-03-02 09:18:57   
137    138           Đỗ Thị Sợi(Hữu)     AP.117  2016-12-13 09:57:14   
146    147              Đỗ Văn Thắng  LM.012.01  2017-02-27 15:06:41   
173    174           Đoàn Thị Phượng     AP.039  2016-12-13 09:51:33   
207    208  Hoàng Văn Chiến ( Thanh)     TG.146  2017-03-11 09:01:16   
208    209             Hoàng Văn Chữ     TG.145  2017-03-11 09:00:45   
215    216       Hoàng Văn Nam (ước)     TG.101  2017-03-10 16:46:20   
216    217  Hoàng Văn Nghĩa ( Tuyết)     TG.134  2017-03-11 08:58:49   
217    218       Hoàng Văn Nhã ( Hào     TG.132  2017-03-11 08:58:14   
376    377         Ngô Thị Dung ( Vỹ     TB.187  2017-02-28 10:25:27   
378    379       Ngô Thị Thảo (ngọc)     TB.051  2017-02-28 10:21:41   
...    ...                       ...        ...                  ...   
1310  1311             Trịnh Văn Đắp     TB.006  2017-02-28 10:10:14   
1311  1312             Trịnh Văn Đậu  TB.042.01  2017-02-28 10:19:52   
1319  1320      Trịnh Văn Dư (luyến)     TG.080  2017-03-10 11:08:12   
1321  1322    Trịnh Văn Dụng ( Tuyến     TB.243  2017-02-24 10:18:47   
1328  1329            Trịnh Văn Hiệu     TB.018  2017-02-28 10:15:40   
1332  1333   Trịnh Văn Hồi ( Lan ) 2     TB.247  2017-02-25 11:09:22   
1337  1338     Trịnh Văn Huân ( Sang  TG.085.01  2017-03-10 16:43:03   
1338  1339             Trịnh Văn Huê  TB.001.05  2017-02-28 10:08:35   
1340  1341       Trinh Văn Huệ ( Tâm     TB.236  2017-02-24 10:00:27   
1347  1348      Trịnh Văn Khởi ( Quý     TB.215  2017-02-28 10:26:37   
1354  1355              Trịnh Văn Lự  TG.093.01  2017-03-10 16:43:43   
1355  1356      Trịnh Văn Luân ( Huệ     TG.127  2017-03-11 08:57:01   
1363  1364    Trịnh Văn Mạnh ( Phong  TB.028.07  2017-02-28 10:17:30   
1367  1368       Trịnh Văn Năm( Phúc     LM.083  2017-02-28 09:51:51   
1376  1377    Trịnh Văn Phương ( ngũ     TB.229  2017-02-17 10:27:18   
1380  1381      Trịnh Văn Quý ( Sạnh     TB.245  2017-02-25 10:58:18   
1383  1384      Trịnh Văn Soi ( Sóng  TB.023.02  2017-02-28 10:16:13   
1385  1386             Trịnh Văn Tâm     TB.003  2017-02-28 10:09:07   
1386  1387     Trịnh Văn Tấn ( bố Tạ     TB.249  2017-03-02 09:10:57   
1393  1394   Trịnh Văn The ( Dung) 2     TB.246  2017-02-25 11:05:11   
1396  1397     Trịnh Văn Thi (chiêm)     TB.017  2017-02-28 10:13:53   
1403  1404            Trịnh Văn Tiền     TB.047  2017-02-28 10:20:30   
1410  1411          Trịnh Văn Trường  TG.122.01  2017-03-11 08:56:25   
1413  1414    Trịnh Văn Tuấn ( Chinh     TB.225

## #3 ONLY HOUSEHOLDS LIVING IN QUANG PHU COMMUNES ARE COUNTED

In [144]:
data=pd.read_excel('CSDL_KhachHang.xlsx')
data

STT                     Tên KH      Mã KH       Ngày nhập CSDL  \
0        1             Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
1        2                    Bà Tiến     CP.022  2017-01-23 12:38:01   
2        3                    Bà Tĩnh     CP.018  2017-01-23 12:38:01   
3        4                     Bà Trừ     CP.056  2017-01-23 12:38:01   
4        5                Bùi Đức Hải     TC.127  2017-01-23 12:38:01   
5        6               Bùi Đức Lĩnh     TC.130  2017-01-23 12:38:01   
6        7            Bùi Quang Thành     TC.155  2017-01-23 12:38:01   
7        8         Bùi Thị Liên(Dũng)     AD.004  2017-01-23 12:38:01   
8        9               Bùi Thị Thao     TC.014  2017-01-23 12:38:01   
9       10               Bùi Văn Đăng     TC.086  2017-01-23 12:38:01   
10      11                Bùi Văn Đạo     TC.045  2017-01-23 12:38:01   
11      12              Bùi Văn Nghĩa     TC.032  2017-01-23 12:38:01   
12      13              Bùi Văn Thang     TC.019  2017-01-23 12:38:01   
13      14              Bùi Văn Thông     TC.018  2017-01-23 12:38:01   
14      15                Bùi Văn Tửu     TC.021  2017-01-23 12:38:01   
15      16        Bùi Văn Tuyên(hinh)     TC.031  2017-01-23 12:38:01   
16      17                Cao Thị Lơi     TB.163  2017-01-23 12:38:01   
17      18         Công Ty Đại Việt 1     LM.052  2017-01-23 12:38:01   
18      19          Công Ty Pháp Việt     LM.053  2017-01-23 12:38:01   
19      20              Đàm Hữu Thành     LM.046  2017-01-23 12:38:01   
20      21                 Đặng Bá Bạ     CP.008  2017-01-23 12:38:01   
21      22            Đặng Quang Nghị     XA.056  2017-01-23 12:38:01   
22      23           Đặng Quang Tuyên     ĐX.014  2017-01-23 12:38:01   
23      24  Đặng Quang Tuyến ( Trang)     ĐX.001  2017-01-23 12:38:01   
24      25               Đặng Thị Bắc     XA.069  2017-01-23 12:38:01   
25      26               Đặng Thị Keo     XA.014  2017-01-23 12:38:01   
26      27             Đặng Thị Tuyên     XA.079  2017-01-23 12:38:01   
27      28           Đặng Trung Tuyến     XA.110  2017-01-23 12:38:01   
28      29                Đặng Văn Bộ  ĐX.025.01  2017-01-23 12:38:01   
29      30              Đặng Văn Động     CP.003  2017-01-23 12:38:01   
...    ...                        ...        ...                  ...   
1428  1429        Trịnh Văn Xuân (xế)     TB.023  2017-01-23 12:38:01   
1429  1430           Trịnh Xuân Chiêm  TB.014.04  2017-01-23 12:38:01   
1430  1431        Trịnh Xuân Mật (Thu     LM.036  2017-01-23 12:38:01   
1431  1432             Trịnh Xuân Sơn     TB.159  2017-02-28 10:22:22   
1432  1433   Trịnh Xuân Trường ( Thoa     TB.177  2017-01-23 12:38:01   
1433  1434      Trường Cấp 2 An Thịnh     AP.162  2017-01-23 12:38:01   
1434  1435      Trường Cấp 2 An Thịnh     AP.163  2017-01-23 12:38:01   
1435  1436  Trường Mầm Non ( Lĩnh Mai  LM.014.01  2017-01-23 12:38:01   
1436  1437  Trường Mầm Non ( Tuyên Bá     TB.016  2016-11-29 11:00:26   
1437  1438     Trường Mầm Non (An Phú     AP.165  2017-01-24 12:58:43   
1438  1439             Trương Thế Phú     TG.095  2017-03-10 16:45:06   
1439  1440              Vũ Công Thắng  LM.047.01  2017-01-23 12:38:01   
1440  1441         Vũ Thị Hằng ( Hùng     LM.109  2016-12-27 13:37:55   
1441  1442                 Vũ Thị Huê     CP.089  2017-01-23 12:38:01   
1442  1443                Vũ Thị Miền     TC.147  2017-01-23 12:38:01   
1443  1444          Vũ Thị Soi ( Chỉ)     AD.036  2017-01-23 12:38:01   
1444  1445                Vũ Thị Thắm  XA.070.01  2017-01-23 12:38:01   
1445  1446                Vũ Thị Thơm     AP.086  2017-01-23 12:38:01   
1446  1447               Vũ Thị Tuyết  TC.135.01  2017-01-23 12:38:01   
1447  1448       Vũ Tiến Cương(tuyết)     ĐX.022  2017-01-23 12:38:01   
1448  1449         Vũ Văn Hưng ( Liên     LM.133  2017-02-23 16:12:48   
1449  1450               Vũ Văn Khang     TC.020  2017-01-23 12:38:01   
1450  1451          Vũ Văn Liên(thời)     ĐX.047  2017-01-23 12:3

In [145]:
data[(data['Xã']=='Quảng Phú')]

STT                             Tên KH      Mã KH       Ngày nhập CSDL  \
0        1                     Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
16      17                        Cao Thị Lơi     TB.163  2017-01-23 12:38:01   
17      18                 Công Ty Đại Việt 1     LM.052  2017-01-23 12:38:01   
18      19                  Công Ty Pháp Việt     LM.053  2017-01-23 12:38:01   
19      20                      Đàm Hữu Thành     LM.046  2017-01-23 12:38:01   
45      46                      Đào Văn Thanh     TG.039  2017-01-23 12:38:01   
49      50               Đinh Huy Cường ( Vải  TG.105.03  2017-01-23 12:38:01   
50      51                      Đinh Huy Dũng  TG.105.02  2017-03-11 08:50:23   
51      52                      Đinh Minh Ngũ     TG.018  2017-01-23 12:38:01   
52      53                     Đinh Thị Hường     TG.141  2017-03-11 09:00:04   
53      54             Đinh Thị Vân Anh (hợi)     LM.079  2017-01-23 12:38:01   
54      55                       Đinh Văn Anh  TG.105.01  2017-03-10 16:48:46   
55      56                       Đinh Văn Ánh     TG.097  2017-01-23 12:38:01   
56      57                 Đinh Văn Ánh ( Đan  TG.107.01  2017-01-23 12:38:01   
57      58                Đinh Văn Ánh ( Đan)  TG.071.06  2017-03-10 11:02:33   
58      59                      Đinh Văn Hanh  TG.076.01  2017-03-10 11:04:28   
59      60                      Đinh Văn Hồng     TG.131  2017-01-23 12:38:01   
60      61                 Đinh Văn Hùng ( Lý  TG.076.04  2017-01-23 12:38:01   
61      62                     Đinh Văn Khang     TG.123  2017-01-23 12:38:01   
62      63                      Đinh Văn Khuê     TG.012  2017-01-23 12:38:01   
63      64                       Đinh Văn Lộc     TG.047  2017-01-23 12:38:01   
64      65                  Đinh Văn Lộc (tý)     TG.001  2017-01-23 12:38:01   
65      66                      Đinh Văn Mạnh  TG.066.01  2017-01-23 12:38:01   
66      67                       Đinh Văn Mật  TG.071.01  2017-01-23 12:38:01   
67      68             Đinh Văn Minh ( Giang)     TG.148  2017-01-23 12:38:01   
68      69                        Đinh Văn Mỡ     TG.066  2017-01-23 12:38:01   
69      70                      Đinh Văn Ngâm     TG.036  2017-01-23 12:38:01   
70      71                      Đinh Văn Phẩm     TG.077  2017-03-10 11:07:01   
71      72            Đinh Văn Phẩm ( Lượt) 2     TG.162  2017-02-23 15:24:19   
72      73                     Đinh Văn Quang     TG.038  2017-01-23 12:38:01   
...    ...                                ...        ...                  ...   
1410  1411                   Trịnh Văn Trường  TG.122.01  2017-03-11 08:56:25   
1411  1412                   Trịnh Văn Trượng  TB.028.04  2017-01-23 12:38:01   
1412  1413                Trịnh Văn Từ ( Thận     TB.161  2017-01-23 12:38:01   
1413  1414             Trịnh Văn Tuấn ( Chinh     TB.225  2017-02-17 09:53:48   
1414  1415              Trịnh Văn Tuấn ( Hưng     TB.173  2017-01-23 12:38:01   
1415  1416                     Trịnh Văn Tuất     TG.098  2017-01-23 12:38:01   
1416  1417                     Trịnh Văn Tươi     TB.028  2017-01-23 12:38:01   
1417  1418               Trịnh Văn Uyên (nõn)     TB.060  2017-01-23 12:38:01   
1418  1419               Trịnh Văn Vận ( Tíu)     TB.070  2017-01-23 12:38:01   
1419  1420               Trịnh Văn Vân (mong)     TG.094  2017-03-10 16:44:31   
1420  1421                 Trịnh Văn Vì ( Vận     TB.235  2017-02-24 09:58:04   
1421  1422             Trịnh Văn Vui ( Nhung)     TB.076  2017-01-23 12:38:01   
1422  1423                Trịnh Văn Xá ( Đăng     TB.185  2017-01-23 12:38:01   
1423  1424               Trịnh Văn Xế ( Quyên     TB.183  2017-01-23 12:38:01   
1424  1425                Trịnh Văn Xê (bách)     TB.025  2017-02-28 10:16:49   
1425  1426             Trịnh Văn Xuân ( Thuận  TB.026.01  2017-01-23 12:38:01   
1426  1427              Trịnh Văn Xuân ( Xế)2  TB.023.04  2017-01-23 12:38:01   
1427  1428  Trịnh văn Xuân (Điền) cháu

In [146]:
QuangPhuCounter=data[(data['Xã']=='Quảng Phú')]
QuangPhuCounter.shape

(688, 15)

### Alright! So you can see there are 688 Households that living in Quang Phu

# #4 Household must have unique household ID

In [147]:
data=pd.read_excel('CSDL_KhachHang.xlsx')
data

STT                     Tên KH      Mã KH       Ngày nhập CSDL  \
0        1             Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
1        2                    Bà Tiến     CP.022  2017-01-23 12:38:01   
2        3                    Bà Tĩnh     CP.018  2017-01-23 12:38:01   
3        4                     Bà Trừ     CP.056  2017-01-23 12:38:01   
4        5                Bùi Đức Hải     TC.127  2017-01-23 12:38:01   
5        6               Bùi Đức Lĩnh     TC.130  2017-01-23 12:38:01   
6        7            Bùi Quang Thành     TC.155  2017-01-23 12:38:01   
7        8         Bùi Thị Liên(Dũng)     AD.004  2017-01-23 12:38:01   
8        9               Bùi Thị Thao     TC.014  2017-01-23 12:38:01   
9       10               Bùi Văn Đăng     TC.086  2017-01-23 12:38:01   
10      11                Bùi Văn Đạo     TC.045  2017-01-23 12:38:01   
11      12              Bùi Văn Nghĩa     TC.032  2017-01-23 12:38:01   
12      13              Bùi Văn Thang     TC.019  2017-01-23 12:38:01   
13      14              Bùi Văn Thông     TC.018  2017-01-23 12:38:01   
14      15                Bùi Văn Tửu     TC.021  2017-01-23 12:38:01   
15      16        Bùi Văn Tuyên(hinh)     TC.031  2017-01-23 12:38:01   
16      17                Cao Thị Lơi     TB.163  2017-01-23 12:38:01   
17      18         Công Ty Đại Việt 1     LM.052  2017-01-23 12:38:01   
18      19          Công Ty Pháp Việt     LM.053  2017-01-23 12:38:01   
19      20              Đàm Hữu Thành     LM.046  2017-01-23 12:38:01   
20      21                 Đặng Bá Bạ     CP.008  2017-01-23 12:38:01   
21      22            Đặng Quang Nghị     XA.056  2017-01-23 12:38:01   
22      23           Đặng Quang Tuyên     ĐX.014  2017-01-23 12:38:01   
23      24  Đặng Quang Tuyến ( Trang)     ĐX.001  2017-01-23 12:38:01   
24      25               Đặng Thị Bắc     XA.069  2017-01-23 12:38:01   
25      26               Đặng Thị Keo     XA.014  2017-01-23 12:38:01   
26      27             Đặng Thị Tuyên     XA.079  2017-01-23 12:38:01   
27      28           Đặng Trung Tuyến     XA.110  2017-01-23 12:38:01   
28      29                Đặng Văn Bộ  ĐX.025.01  2017-01-23 12:38:01   
29      30              Đặng Văn Động     CP.003  2017-01-23 12:38:01   
...    ...                        ...        ...                  ...   
1428  1429        Trịnh Văn Xuân (xế)     TB.023  2017-01-23 12:38:01   
1429  1430           Trịnh Xuân Chiêm  TB.014.04  2017-01-23 12:38:01   
1430  1431        Trịnh Xuân Mật (Thu     LM.036  2017-01-23 12:38:01   
1431  1432             Trịnh Xuân Sơn     TB.159  2017-02-28 10:22:22   
1432  1433   Trịnh Xuân Trường ( Thoa     TB.177  2017-01-23 12:38:01   
1433  1434      Trường Cấp 2 An Thịnh     AP.162  2017-01-23 12:38:01   
1434  1435      Trường Cấp 2 An Thịnh     AP.163  2017-01-23 12:38:01   
1435  1436  Trường Mầm Non ( Lĩnh Mai  LM.014.01  2017-01-23 12:38:01   
1436  1437  Trường Mầm Non ( Tuyên Bá     TB.016  2016-11-29 11:00:26   
1437  1438     Trường Mầm Non (An Phú     AP.165  2017-01-24 12:58:43   
1438  1439             Trương Thế Phú     TG.095  2017-03-10 16:45:06   
1439  1440              Vũ Công Thắng  LM.047.01  2017-01-23 12:38:01   
1440  1441         Vũ Thị Hằng ( Hùng     LM.109  2016-12-27 13:37:55   
1441  1442                 Vũ Thị Huê     CP.089  2017-01-23 12:38:01   
1442  1443                Vũ Thị Miền     TC.147  2017-01-23 12:38:01   
1443  1444          Vũ Thị Soi ( Chỉ)     AD.036  2017-01-23 12:38:01   
1444  1445                Vũ Thị Thắm  XA.070.01  2017-01-23 12:38:01   
1445  1446                Vũ Thị Thơm     AP.086  2017-01-23 12:38:01   
1446  1447               Vũ Thị Tuyết  TC.135.01  2017-01-23 12:38:01   
1447  1448       Vũ Tiến Cương(tuyết)     ĐX.022  2017-01-23 12:38:01   
1448  1449         Vũ Văn Hưng ( Liên     LM.133  2017-02-23 16:12:48   
1449  1450               Vũ Văn Khang     TC.020  2017-01-23 12:38:01   
1450  1451          Vũ Văn Liên(thời)     ĐX.047  2017-01-23 12:3

In [150]:
data['Tên KH'].duplicated().sum()

49

### So you can see there are 49 Households that have the same ID. Therefore i'm gonna remove all duplicate rows 

In [151]:
data.shape

(1458, 15)

In [152]:
data.drop_duplicates(subset=['Tên KH']).shape

(1409, 15)

### =>>> okay, 1458-1409=49 which is equal the number of duplicated rows

In [153]:
data

STT                     Tên KH      Mã KH       Ngày nhập CSDL  \
0        1             Bà Liên ( Chức  TG.095.01  2017-03-10 16:45:43   
1        2                    Bà Tiến     CP.022  2017-01-23 12:38:01   
2        3                    Bà Tĩnh     CP.018  2017-01-23 12:38:01   
3        4                     Bà Trừ     CP.056  2017-01-23 12:38:01   
4        5                Bùi Đức Hải     TC.127  2017-01-23 12:38:01   
5        6               Bùi Đức Lĩnh     TC.130  2017-01-23 12:38:01   
6        7            Bùi Quang Thành     TC.155  2017-01-23 12:38:01   
7        8         Bùi Thị Liên(Dũng)     AD.004  2017-01-23 12:38:01   
8        9               Bùi Thị Thao     TC.014  2017-01-23 12:38:01   
9       10               Bùi Văn Đăng     TC.086  2017-01-23 12:38:01   
10      11                Bùi Văn Đạo     TC.045  2017-01-23 12:38:01   
11      12              Bùi Văn Nghĩa     TC.032  2017-01-23 12:38:01   
12      13              Bùi Văn Thang     TC.019  2017-01-23 12:38:01   
13      14              Bùi Văn Thông     TC.018  2017-01-23 12:38:01   
14      15                Bùi Văn Tửu     TC.021  2017-01-23 12:38:01   
15      16        Bùi Văn Tuyên(hinh)     TC.031  2017-01-23 12:38:01   
16      17                Cao Thị Lơi     TB.163  2017-01-23 12:38:01   
17      18         Công Ty Đại Việt 1     LM.052  2017-01-23 12:38:01   
18      19          Công Ty Pháp Việt     LM.053  2017-01-23 12:38:01   
19      20              Đàm Hữu Thành     LM.046  2017-01-23 12:38:01   
20      21                 Đặng Bá Bạ     CP.008  2017-01-23 12:38:01   
21      22            Đặng Quang Nghị     XA.056  2017-01-23 12:38:01   
22      23           Đặng Quang Tuyên     ĐX.014  2017-01-23 12:38:01   
23      24  Đặng Quang Tuyến ( Trang)     ĐX.001  2017-01-23 12:38:01   
24      25               Đặng Thị Bắc     XA.069  2017-01-23 12:38:01   
25      26               Đặng Thị Keo     XA.014  2017-01-23 12:38:01   
26      27             Đặng Thị Tuyên     XA.079  2017-01-23 12:38:01   
27      28           Đặng Trung Tuyến     XA.110  2017-01-23 12:38:01   
28      29                Đặng Văn Bộ  ĐX.025.01  2017-01-23 12:38:01   
29      30              Đặng Văn Động     CP.003  2017-01-23 12:38:01   
...    ...                        ...        ...                  ...   
1428  1429        Trịnh Văn Xuân (xế)     TB.023  2017-01-23 12:38:01   
1429  1430           Trịnh Xuân Chiêm  TB.014.04  2017-01-23 12:38:01   
1430  1431        Trịnh Xuân Mật (Thu     LM.036  2017-01-23 12:38:01   
1431  1432             Trịnh Xuân Sơn     TB.159  2017-02-28 10:22:22   
1432  1433   Trịnh Xuân Trường ( Thoa     TB.177  2017-01-23 12:38:01   
1433  1434      Trường Cấp 2 An Thịnh     AP.162  2017-01-23 12:38:01   
1434  1435      Trường Cấp 2 An Thịnh     AP.163  2017-01-23 12:38:01   
1435  1436  Trường Mầm Non ( Lĩnh Mai  LM.014.01  2017-01-23 12:38:01   
1436  1437  Trường Mầm Non ( Tuyên Bá     TB.016  2016-11-29 11:00:26   
1437  1438     Trường Mầm Non (An Phú     AP.165  2017-01-24 12:58:43   
1438  1439             Trương Thế Phú     TG.095  2017-03-10 16:45:06   
1439  1440              Vũ Công Thắng  LM.047.01  2017-01-23 12:38:01   
1440  1441         Vũ Thị Hằng ( Hùng     LM.109  2016-12-27 13:37:55   
1441  1442                 Vũ Thị Huê     CP.089  2017-01-23 12:38:01   
1442  1443                Vũ Thị Miền     TC.147  2017-01-23 12:38:01   
1443  1444          Vũ Thị Soi ( Chỉ)     AD.036  2017-01-23 12:38:01   
1444  1445                Vũ Thị Thắm  XA.070.01  2017-01-23 12:38:01   
1445  1446                Vũ Thị Thơm     AP.086  2017-01-23 12:38:01   
1446  1447               Vũ Thị Tuyết  TC.135.01  2017-01-23 12:38:01   
1447  1448       Vũ Tiến Cương(tuyết)     ĐX.022  2017-01-23 12:38:01   
1448  1449         Vũ Văn Hưng ( Liên     LM.133  2017-02-23 16:12:48   
1449  1450               Vũ Văn Khang     TC.020  2017-01-23 12:38:01   
1450  1451          Vũ Văn Liên(thời)     ĐX.047  2017-01-23 12:3